File: Eeasy/1_TwoSum.java
import java.util.HashMap;
import java.util.Map;

public class TwoSum {

    public static int[] twoSum(int[] nums, int target) {
        Map<Integer, Integer> numIndexMap = new HashMap<>();

        for (int i = 0; i < nums.length; i++) {
            int complement = target - nums[i];

            if (numIndexMap.containsKey(complement)) {
                return new int[]{numIndexMap.get(complement), i};
            }

            numIndexMap.put(nums[i], i);
        }

        throw new IllegalArgumentException("No two sum solution");
    }

    public static void main(String[] args) {
        // Example usage:
        int[] nums = {2, 7, 11, 15};
        int target = 9;

        int[] result = twoSum(nums, target);

        System.out.println("Indices of the two numbers: " + result[0] + ", " + result[1]);
    }
}

File: easy/1_two_sum.cs
public class Solution {
    public int[] TwoSum(int[] nums, int target) {
        // Initialize a dictionary to store numbers and their indices
        Dictionary<int, int> numToIndex = new Dictionary<int, int>();
        
        // Loop through each number in the array
        for (int i = 0; i < nums.Length; i++) {
            // Calculate the complement needed to reach the target
            int complement = target - nums[i];
            
            // Check if the complement is in the dictionary
            if (numToIndex.ContainsKey(complement)) {
                // If found, return the indices of the current number and the complement
                return new int[] { numToIndex[complement], i };
            }
            
            // If complement not found, add the current number and its index to the dictionary
            numToIndex[nums[i]] = i;
        }
        
        // The problem guarantees a solution exists, so we shouldn't reach here
        throw new ArgumentException("No two sum solution");
    }
}
